jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -673,11 +684,109 @@ public void write(int b) throws IOException
 
     @Override
     public void print(String s) throws IOException
+    {
+        print(s,false);
+    }
+
+    @Override
+    public void println(String s) throws IOException
+    {
+        print(s,true);
+    }
+
+    private void print(String s, boolean eoln) throws IOException
     {
         if (isClosed())
             throw new IOException("Closed");
 
-        write(s.getBytes(_channel.getResponse().getCharacterEncoding()));
+        // TODO would a Threadlocal pool be better?
+        String charset = _channel.getResponse().getCharacterEncoding();
+        if (_encoder==null || !_encoder.charset().name().equals(charset))
+            _encoder = Charset.forName(charset).newEncoder();
+        else
+            _encoder.reset();
+
+        CharBuffer in = CharBuffer.wrap(s);
+        CharBuffer crlf = eoln?CharBuffer.wrap("\r\n"):null;
+        ByteBuffer out = getHttpChannel().getByteBufferPool().acquire((int)(1+(s.length()+2)*_encoder.averageBytesPerChar()),false);
```

Comment:
Considering that the `ByteBufferPool` returns buffers that have capacity a multiple of (typically) some power of 2, I would avoid the rigmarole of guessing the size here, and possibly grow the buffer below.
Use `encoder.maxBytesPerChar()` plus some extra for the CRLF (why there is the `1+` at the beginning?).

Reply:
No growing. Just go for the max plus some extra. Should be much simpler.

Reply:
ok

Reply:
Nah - I decided to stay with growing.  The code is written, it's not that complex and it can avoid a 4* or 6* blow out of buffer usage.  It also follows the example of how the JVM implements it's own internal version of this.